game.py

```diff
@@ -684,6 +686,36 @@ def _circle(self, r, cx, cy):
     def _footer(self):
         return '</svg>\n'
 
+    def save_highscore(self):
```

Comment:
I think you can easily implement the `read_file` and `write_file` methods in `RecallActivity.py` as you can set the scores whenever you need to and don't bother about calling either function as they get called automatically, you can set a `highscore` here which can be accessed with `self._game.highscore` and use in both methods.

Reply:
I tried using it initially, but then I realised `high score` needs to be fetched during the new instance of the activity and not when an activity is resumed. The `read_file` is called only when an activity is resumed. Hence I had to explicitly call a function in each instance of an activity.

Reply:
Yeah that makes perfect sense, something similar was implemented in [dimensions](https://github.com/sugarlabs/dimensions/commit/4f5a9e6c7d64fd26b259d8ec4f1381e27cf641c8) you might want to take a look at `Dimensions.py` and `game.py` of said commit.